### PR TITLE
Feature/4 add put endpoint to update a ticket

### LINKED ISSUE #4: Add PUT endpoint to update a ticket
**Description**
As a user
I want to update tickets.

**To-Do**
- [x] Write a put endpoint
- [x] Write tests for put endpoint
- [x] Write a frontend to edit the ticket desription

**Frontend**
- When clicking on a card, the sidepanel opens and the ticket is displayed there
- above the ticket is an edit button which makes the ticket editable

### PR DESCRIPTION
**Pull request by jonashonecker**
* Add PUT endpoint in backend
* Add input validation
* Add unit and integration tests
* Add frontend

Closes #4

### CODE REVIEW

File: backend/src/main/java/com/github/jonashonecker/backend/ticket/domain/UpdateTicket.java
Comment (gcode-de) [inline backend/src/main/java/com/github/jonashonecker/backend/ticket/domain/UpdateTicket.java:1]: This looks like a DTO. Should it have "DTO" in the name?

File: frontend/src/components/buttons/UpdateButton.tsx
Comment (gcode-de) [inline frontend/src/components/buttons/UpdateButton.tsx:1]: don't you want o use a single but flexible component for all those buttons?

**Review by gcode-de — APPROVED** [review]
nice!!

File: backend/src/main/java/com/github/jonashonecker/backend/ticket/domain/UpdateTicket.java
Comment (jonashonecker) [reply backend/src/main/java/com/github/jonashonecker/backend/ticket/domain/UpdateTicket.java:1]: Good point, thank you. 👍  (Adjusted in latest commit)

File: frontend/src/components/buttons/UpdateButton.tsx
Comment (jonashonecker) [reply frontend/src/components/buttons/UpdateButton.tsx:1]: I think I prefer less logic in components and therefore have a little more repetetiveness. But some of them could be easily grouped, that's right. I will save that for later. 👍

**Pull request merged**